packages/dds/ordered-collection/src/test/fuzzUtils.ts

```diff
@@ -0,0 +1,233 @@
+	type: "resolve";
+	result: ConsensusResult;
+}
+
+/**
+ * Represents ConsensusOrderedCollection operation types for fuzz testing
+ */
+export type ConsensusOrderedCollectionOperation =
+	| AddOperation
+	| AcquireOperation
+	| ResolveOperation;
+
+function makeOperationGenerator(): Generator<
+	ConsensusOrderedCollectionOperation,
+	FuzzTestState
+> {
+	type OpSelectionState = FuzzTestState & {
+		itemValue: string;
+	};
+
+	const valuePoolRandom = makeRandom(0);
+	const dedupe = <T>(arr: T[]): T[] => [...new Set(arr)];
+	const valuePool = dedupe(
+		Array.from({ length: valueConfigs.valuePoolSize }, () =>
+			valuePoolRandom.string(valueConfigs.valueStringLength),
+		),
+	);
+
+	async function add(state: OpSelectionState): Promise<AddOperation> {
+		return {
+			type: "add",
+			value: state.itemValue,
+		};
+	}
+
+	async function acquire(state: OpSelectionState): Promise<AcquireOperation> {
+		return {
+			type: "acquire",
+		};
+	}
+
+	async function resolve(state: OpSelectionState): Promise<ResolveOperation> {
+		return {
+			type: "resolve",
+			result: state.random.pick([ConsensusResult.Complete, ConsensusResult.Release]),
+		};
+	}
+
+	const clientBaseOperationGenerator = createWeightedGenerator<
+		ConsensusOrderedCollectionOperation,
+		OpSelectionState
+	>([
+		[add, 1],
+		[acquire, 1],
+		[resolve, 1],
+	]);
+
+	return async (state: FuzzTestState) =>
+		clientBaseOperationGenerator({
+			...state,
+			itemValue: state.random.pick(valuePool),
+		});
+}
+
+function makeReducer(): Reducer<ConsensusOrderedCollectionOperation, FuzzTestState> {
+	const pendingCallbacks = new Map<string, string>();
+	const callbackResolver = new EventEmitter();
+
+	const reducer = combineReducers<ConsensusOrderedCollectionOperation, FuzzTestState>({
+		add: ({ client }, { value }) => {
+			client.channel.add(value).catch((error) => {
+				throw error;
+			});
+		},
+		acquire: ({ client }) => {
+			const callback = async (value: string): Promise<ConsensusResult> => {
+				const callbackId = uuid();
+				pendingCallbacks.set(callbackId, value);
+				return new Promise<ConsensusResult>((resolve) => {
+					const onCallbackResolve = (id: string, result: ConsensusResult): void => {
+						if (id === callbackId) {
+							callbackResolver.off("resolve", onCallbackResolve);
+							resolve(result);
+						}
+					};
+					callbackResolver.on("resolve", onCallbackResolve);
+				});
+			};
+			client.channel.acquire(callback).catch((error) => {
+				throw error;
+			});
+		},
+		resolve: ({ client }, { result }) => {
+			new Promise<string>((resolve) => {
+				if (result === ConsensusResult.Complete) {
+					client.channel.once("localComplete", (value: string) => {
+						resolve(value);
+					});
+				} else {
+					client.channel.once("localRelease", (value: string) => {
+						resolve(value);
+					});
+				}
+			})
+				.then((resolvedValue: string) => {
+					const [callbackId, value] = pendingCallbacks.entries().next().value as [
```

Comment:
Probably want to pick an item at random to resolve rather than just the next one?  Maybe doesn't matter that much though.